File: Week-1/mario-less/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int height;
    // Prompt user for height, ensuring it's a positive integer between 1 and 8 inclusive
    do
    {
        height = get_int("Height: ");
    } while (height < 1 || height > 8);

    // Generate pyramid of hashes
    for (int i = 0; i < height; i++)
    {
        // Print spaces before the hashes
        for (int j = 0; j < height - i - 1; j++)
        {
            printf(" ");
        }

        // Print hashes
        for (int j = 0; j < i + 1; j++)
        {
            printf("#");
        }

        // Print newline
        printf("\n");
    }

    return 0;
}

File: Week-6/Sentimental-Mario-less/mario.py
# Prompt user for height, ensuring it's a positive integer between 1 and 8 inclusive
while True:
    try:
        height = int(input("Height: "))
        if 1 <= height <= 8:
            break
        else:
            print("Height must be between 1 and 8 inclusive.")
    except ValueError:
        print("Invalid input. Please enter an integer.")

# Generate pyramid of hashes
for i in range(height):
    # Print spaces before the hashes
    for j in range(height - i - 1):
        print(" ", end="")

    # Print hashes
    for j in range(i + 1):
        print("#", end="")

    # Print newline
    print()
